streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1121,13 +1122,31 @@ public String name() {
 
             @Override
             public void apply(final StreamTask task) {
-                task.suspend();
+                try {
+                    task.suspend();
+                } catch (final Exception e) {
+                    try {
+                        task.close(false);
+                    } catch (final Exception f) {
+                        log.error("{} Closing task {} failed: ", logPrefix, task.id, f);
+                    }
+                    throw e;
```

Comment:
I am actually not sure, if we need to fail here. As a task was revoked, we could also just close it, and don't report back as previously active task (at least for eos as the state is corrupted) -- for non-eos, it might not make a difference. However, it seems we could just carry on the rebalance. But I am not 100% sure. Same for line 1148. Please comment on this.